scene/animation/animation_tree.cpp

```diff
@@ -764,25 +765,27 @@ void AnimationTree::_setup_animation_player() {
 		return;
 	}
 
-	AnimationMixer *mixer = Object::cast_to<AnimationMixer>(get_node_or_null(animation_player));
-	if (mixer) {
-		if (!mixer->is_connected(SNAME("caches_cleared"), callable_mp(this, &AnimationTree::_setup_animation_player))) {
-			mixer->connect(SNAME("caches_cleared"), callable_mp(this, &AnimationTree::_setup_animation_player), CONNECT_DEFERRED);
+	// Using AnimationPlayer here is for compatibility. Changing to AnimationMixer needs extra work like error handling.
+	// In future 5.0, it should be possible to choose whether to import Animation as Tree or Player in the import settings.
```

Comment:
We shouldn't mention Godot 5 or anything like that, because there are currently no plans for that. You can probably just drop the second line.